packages/shared/src/components/plus/PlusIOS.tsx

```diff
@@ -59,7 +59,11 @@ export const PlusIOS = ({
     promisifyEventListener<void, 'true' | 'false'>(
       'iap-loading',
       ({ detail }) => {
-        setIsLoading(detail.toLowerCase() === 'true');
+        try {
+          setIsLoading(detail.toLowerCase() === 'true');
+        } catch {
+          setIsLoading(false);
+        }
```

Comment:
`setIsLoading` won't throw, I think wrapping `await promisifyEventListener` should cover some cases

Reply:
It could if `detail` isn't a string

Reply:
let's handle that specifically then?

Reply:
I created a util function that should make sure it is a string